### PR TITLE
Subscribe to events when starting a process instance anywhere

### LINKED ISSUE #9391: Subscribe to events of created element scopes
Subscribe to the events of the created elements scopes:
- subscribe each flow scope that is created as part of #9390 to the messages and timers using the existing logic (i.e. `CatchEventBehavior.subscribeToEvents`). 
- flush the side effect queue, to actually subscribe (necessary for both messages as well timer events).

Watch out to not create event subscriptions multiple times when we create a process instance with multiple tokens.

[See spike](https://github.com/camunda/zeebe/commit/b2cb74629694dfc059bb1cdb91ae3a263d0b0050#diff-993ccf60e85a2003a595381edf26e3cad5c69aa2f3d9d89e658911b0851081f5R105-R113) as example.

Blocked by #9390

### PR DESCRIPTION
**Pull request by saig0**
## Description

* reuse the existing logic to subscribe to events (i.e. to create event subscriptions)
* create the event subscriptions for the process and the flow scope of the given elements
* test the creation of the event subscriptions indirectly by verifying that the event subprocess / boundary event is triggered. This way, we verify that the events can also be triggered without repeating the code.

## Related issues

closes #9391 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Review by pihme — APPROVED** [review]
Looks good to me. Nice job!

File: engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/CreateProcessInstanceProcessor.java [hunk 1]
```diff
@@ -245,6 +258,9 @@ private void activateElementsForStartInstructions(
           commandWriter.appendFollowUpCommand(
               elementInstanceKey, ProcessInstanceIntent.ACTIVATE_ELEMENT, elementRecord);
         });
+
+    // applying the side effects is part of creating the event subscriptions
+    sideEffectQueue.flush();
```
[the remark this answers is not in the document]
Comment (saig0) [reply engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/CreateProcessInstanceProcessor.java:263]: Yes :smile: Side-effects are not 100% clear yet. We will see :rocket:

**Comment by saig0** [comment]
@pihme thank you for the fast review :+1:  

bors merge

**Comment by saig0** [comment]
bors merge

**Comment by saig0** [comment]
bors merge

**Pull request merged**